scripts/clear-signed-agreements.py

```diff
@@ -52,22 +50,29 @@ def main(stage, framework_slug, api_token, user, supplier_ids=None):
     suppliers = api_client.find_framework_suppliers(framework_slug, agreement_returned=True)['supplierFrameworks']
 
     if supplier_ids is not None:
-        missing_supplier_ids = set(supplier_ids) - set(supplier['supplierId'] for supplier in suppliers)
-        if missing_supplier_ids:
-            raise Exception("Invalid supplier IDs: {}".format(', '.join(str(x) for x in missing_supplier_ids)))
-    else:
-        supplier_ids = set(supplier['supplierId'] for supplier in suppliers)
-
-    for supplier_id in supplier_ids:
-        logger.info("Resetting agreement returned flag for supplier {supplier_id}",
-                    extra={'supplier_id': supplier_id})
-        api_client.unset_framework_agreement_returned(supplier_id, framework_slug, user)
+        suppliers = [
+            supplier for supplier in suppliers
+            if supplier['supplierId'] in supplier_ids
+            and (not cutoff_time or supplier['agreementReturnedAt'] < cutoff_time)
+        ]
+        supplier_ids = [supplier['supplierId'] for supplier in suppliers]
+
+    for supplier in suppliers:
+        logger.info("Resetting agreement returned flag for supplier {supplier_id} returned at {returned_at}",
+                    extra={'supplier_id': supplier['supplierId'], 'returned_at': supplier['agreementReturnedAt']})
+        api_client.unset_framework_agreement_returned(supplier['supplierId'], framework_slug, user)
 
     signed_agreements = filter(
-        lambda x: match_signed_agreements(supplier_ids, x['path']),
+        lambda x: re.search(r'/(\d+)-signed-framework-agreement.', x['path']),
         agreements_bucket.list('{}/agreements/'.format(framework_slug))
     )
 
+    if supplier_ids is not None:
+        signed_agreements = [
+            agreement for agreement in signed_agreements
+            if int(re.search(r'/(\d+)-signed-framework-agreement', agreement['path']).group(1)) in supplier_ids
```

Comment:
No problems anymore! Not sure what I was getting originally, but it was probably a side effect of something else.